jsonrpc/ws-connection/src/ws.ts

```diff
@@ -121,32 +123,26 @@ export class WsConnection implements IJsonRpcConnection {
     return new Promise((resolve, reject) => {
       const opts = !isReactNative() ? { rejectUnauthorized: !isLocalhostUrl(url) } : undefined;
       const socket: WebSocket = new WS(url, [], opts);
+      if (isBrowser()) {
+        socket.onerror = (event: Event) => {
+          const errorEvent = event as ErrorEvent;
+          reject(this.emitError(errorEvent.error));
+        };
+      } else {
+        (socket as any).on("error", (errorEvent: any) => {
```

Comment:
Thanks for fixing this in the `ws`-facing scenario 🙌 

What I don't understand right now: how come e.g. `socket.onopen` still apparently works with (node) `ws` if the listener syntax seems to be `socket.on("<eventType>")` 🤔

Reply:
https://github.com/websockets/ws/blob/master/lib/websocket.js#L578

`ws` also registers the `onopen,onmessage` etc but the error handling is captured only via `on.("error")` or atleast in the `socket hang up` scenario